n2v/grid/grider.py

```diff
@@ -0,0 +1,648 @@
+
+        Returns
+        -------
+        dft_grid: list
+            Numpy arrays corresponding to x,y,z, and w.
+            Shape: (4, npoints**3)
+        
+        """
+
+        try:
+            vpot = wfn.V_potential()
+        except:
+            raise ValueError("Wfn object does not contain a Vpotential object. Try with one obtained from a DFT calculation")
+
+        nblocks = vpot.nblocks()
+        blocks = [vpot.get_block(i) for i in range(nblocks)]
+        npoints = vpot.grid().npoints()
+        points_function = vpot.properties()[0]
+
+        x = np.empty((npoints))
+        y = np.empty((npoints))
+        z = np.empty((npoints))
+        w = np.empty((npoints))
+
+        offset = 0
+        for i_block in blocks:
+            b_points = i_block.npoints()
+            offset += b_points
+
+            x[offset - b_points : offset] = i_block.x().np
+            y[offset - b_points : offset] = i_block.x().np
+            z[offset - b_points : offset] = i_block.x().np
+            w[offset - b_points : offset] = i_block.x().np
+
+        dft_grid = [x,y,z,w]
+
+        return dft_grid
+
+
+    #Quantities on Grid
+    def on_grid_ao(self, coeff, grid=None, vpot=None, cubic_grid=False):
+        """
+        Generates a quantity on the grid given its ao representation
+
+        Parameters
+        ----------
+        coeff: np.ndarray
+            Vector/Matrix of quantity on ao basis. Shape: {(num_ao_basis, ), (num_ao_basis, num_ao_basis)}
+        grid: np.ndarray. Shape ()
+            Grid where density will be computed.
+        cubic_grid: bool    
+            If False the resulting array won't be reshaped.
+            If True the resulting array will be reshaped as (npoints, nponits, npoints). 
+            Where npoints is the number of points for the grid in any one dimension.
+        vpot: psi4.core.VBase
+            Vpotential object with info about grid. 
+            Provides DFT spherical grid. Only comes to play if no grid is given. 
+
+        Returns
+        -------
+        coeff_r: np.ndarray
+            Quantity expressed by the coefficient on the given grid 
+            Shape: (npoints) if cubic_grid is False
+                   (npoints, npoints, npoints) if cubic_grid is True
+
+        """
+
+        if cubic_grid is True and grid is not None:
+            nshape = grid.shape[1]
+            grid, shape = self.generate_mesh(grid)
+        elif cubic_grid is True and grid is None:
+            raise ValueError("'Cubic Grid' requires to explicitly specify grid")
+
+        if grid is not None and vpot is None:
+            blocks, npoints, points_function = self.grid_to_blocks(grid)
+        elif grid is None and vpot is not None:
+            nblocks = vpot.nblocks()
+            blocks = [vpot.get_block(i) for i in range(nblocks)]
+            npoints = vpot.grid().npoints()
+            points_function = vpot.properties()[0]
+        elif grid is None and vpot is None:
```

Comment:
Could you change this line just to "else:" to handle cases both None and both given?